### PR TITLE
Handling for external displays + Steam control for max mode

### PR DESCRIPTION
**Pull request by Joshua-Ashton**

### CODE REVIEW

File: src/wlserver.cpp [hunk 1]
```diff
@@ -1060,3 +1065,24 @@ void wlserver_surface_finish( struct wlserver_surface *surf )
...
+{
+	gamescope_xwayland_server_t *server = wlserver_get_xwayland_server( idx );
+	if ( !server )
+		return;
+	wl_client *client = server->get_client();
+
+	struct wl_resource *resource;
+	wl_resource_for_each( resource, &wlserver.wlr.output->resources )
+	{
+		struct wlr_output *output = wlr_output_from_resource( resource );
```
Comment (emersion) [inline src/wlserver.cpp:1079]: This is the same as `wlserver.wlr.output`

File: src/wlserver.cpp [hunk 2]
```diff
@@ -1060,3 +1065,24 @@ void wlserver_surface_finish( struct wlserver_surface *surf )
...
+		if ( output && wl_resource_get_client( resource ) == client )
+		{
+			wl_output_send_mode( resource, WL_OUTPUT_MODE_CURRENT, w, h, refresh * 1000 );
+			wl_output_send_done( resource );
+			wl_signal_emit( &output->events.mode, output );
```
Comment (emersion) [inline src/wlserver.cpp:1084]: Let's not do this:

- We haven't setup any mode change listener
- Compositors aren't supposed to manually emit this signal

Comment (Joshua-Ashton) [reply src/wlserver.cpp]: Done.

File: src/wlserver.cpp (hunk 1 above)
Comment (Joshua-Ashton) [reply src/wlserver.cpp]: Fixed.

**Pull request merged**